src/std-indices/STDIndicesStream.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// For full license terms please see the LICENSE file distributed with this
+// source code
+
+#pragma once
+
+#include <iostream>
+#include <stdexcept>
+#include "Stream.h"
+
+#define IMPLEMENTATION_STRING "STD (index-oriented)"
+
+template <typename N>
+class ranged {
```

Comment:
Can you add some high level comments explaining what this is? I.e it’s a lightweight counting iterator which will be used by the STL algorithms …..